lib/index.js

```diff
@@ -5,17 +5,22 @@
 var Dialog = require('./Dialog');
 module.exports = function Conversation(bot) {
 
-    var _talkingTo = {},//TODO: Use the robot brain for this, it will probably be more scalable.
-        spy = function (msg) {
+    var _talkingTo = {};//TODO: Use the robot brain for this, it will probably be more scalable.
+
+    //Register a custom listener that will spy on all incoming messages
+    bot.listen(
+        function matcher(msg) {
+            return true; //All messages are accepted.
```

Comment:
This is still going to have the same behavior (and problems) as `robot.hear /.*/`, buuut what if you instead only return true when the message matches one of the choices? That would behave properly (matcher returns true iff the message is going to get processed).